pkg/store/store.go

```diff
@@ -162,16 +207,55 @@ func (s *atomixStore) Get(ctx context.Context, id topoapi.ID) (*topoapi.Object,
 	if err != nil {
 		return nil, errors.FromAtomix(err)
 	}
-	return decodeObject(*entry)
+	obj, err := decodeObject(*entry)
+	if err != nil {
+		return nil, err
+	}
+	s.addSrcTgts(obj)
+	return obj, nil
 }
 
 func (s *atomixStore) Delete(ctx context.Context, id topoapi.ID) error {
 	if id == "" {
 		return errors.NewInvalid("ID cannot be empty")
 	}
 
+	s.relations.srcLock.Lock()
```

Comment:
Definitely split this into its own method. Delete should be pretty much as it was; maybe with addition of one or two lines.